client/src/components/PeopleView/PeopleView.js

```diff
@@ -24,6 +24,52 @@ export default class PeopleView extends Component {
     handleClick = (e, p) => this.setState({isShowingModal: true, selectedEmployee: e})
     handleClose = () => this.setState({isShowingModal: false})
 
+    componentDidMount() {
+        const { match, entryActions } = this.props;
+
+        this.weekChanger = setInterval(() => {
+            const week = this.props.date.week();
+            const weekNow = moment().week();
+
+            if (!match.params.week && week < weekNow) {
+                entryActions.setWeek(week);
+            }
+        }, 60000);
+    }
+    componentWillUnmount() {
+        clearInterval(this.weekChanger);
+    }
+    componentWillMount() {
+        const { match, entryActions } = this.props;
+
+        const week = match.params.week;
+        if (week) {
+            entryActions.setWeek(week);
+        }
+    }
+    componentWillReceiveProps(nextProps) {
+        const { match, entryActions } = nextProps;
+        const { date } = this.props;
+
+        const week = match.params.week;
+        const weekNumber = nextProps.date.week();
+        const weekNumberNow = moment().week();
+
+        if (this.props.d !== nextProps.d && week) {
+            entryActions.setWeek(week);
+        }
+
+        if (date === nextProps.date) {
+            return;
+        }
+    
+        if (weekNumber !== weekNumberNow) {
```

Comment:
Starting from this line, you could probably have an own method for the rest, `updatePath()` or something. Would make it perhaps a little less convoluted (especially since you don't need `weekNumber` and `weekNumberNow` until this).